File: app/src/main/java/com/mahendri/pasbeli/ui/history/HistoryAdapter.java
package com.mahendri.pasbeli.ui.history;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.mahendri.pasbeli.R;
import com.mahendri.pasbeli.entity.BarangHarga;
import com.mahendri.pasbeli.entity.HargaKonsumen;

import java.text.SimpleDateFormat;
import java.util.Calendar;
import java.util.Date;
import java.util.List;
import java.util.Locale;

/**
 * @author Mahendri Dwicahyo
 */

public class HistoryAdapter extends RecyclerView.Adapter<HistoryAdapter.HistoryHolder> {

    private List<BarangHarga> data;

    HistoryAdapter(List<BarangHarga> data) {
        this.data = data;
    }

    @Override
    public HistoryHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        Context context = parent.getContext();
        View v = LayoutInflater.from(context).inflate(R.layout.item_komoditi, parent, false);
        return new HistoryHolder(v);
    }

    @Override
    public void onBindViewHolder(HistoryHolder holder, int position) {
        BarangHarga barang = data.get(position);
        holder.setData(barang);
    }

    @Override
    public int getItemCount() {
        if (data == null) return 0;
        return data.size();
    }

    void swapData(List<BarangHarga> newData) {
        if (newData != null) {
            data = newData;
            notifyDataSetChanged();
        }
    }

    class HistoryHolder extends RecyclerView.ViewHolder {

        private TextView dataText;
        private TextView locationText;
        private TextView timeText;

        HistoryHolder(View itemView) {
            super(itemView);
            dataText = itemView.findViewById(R.id.komoditi_data);
            locationText = itemView.findViewById(R.id.komoditi_tempat);
            timeText = itemView.findViewById(R.id.komoditi_waktu);
        }

        void setData (BarangHarga barangHarga) {
            dataText.setText(String.format("Mendata %s seharga Rp %s",
                    barangHarga.getNama(), barangHarga.getHarga()));
            locationText.setText(barangHarga.getNamaTempat());
            timeText.setText(convertStringDate(barangHarga.getWaktuCatat()));
        }

        String convertStringDate(long time){
            Calendar c = Calendar.getInstance();

            // set the calendar to start of today
            c.set(Calendar.HOUR_OF_DAY, 0);
            c.set(Calendar.MINUTE, 0);
            c.set(Calendar.SECOND, 0);
            c.set(Calendar.MILLISECOND, 0);
            long today = c.getTimeInMillis();

            if(time > today){
                SimpleDateFormat formatDate = new SimpleDateFormat("HH:mm", Locale.getDefault());
                return formatDate.format(new Date(time));
            } else {
                SimpleDateFormat formatDate = new SimpleDateFormat("dd MMM", Locale.getDefault());
                return formatDate.format(new Date(time));
            }
        }

    }
}

File: app/src/main/java/com/mahendri/pasbeli/ui/history/HistoryAdapter.kt
package com.mahendri.pasbeli.ui.history

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.ViewGroup
import com.mahendri.pasbeli.databinding.ItemKomoditiBinding
import com.mahendri.pasbeli.entity.BarangHarga

/**
 * @author Mahendri Dwicahyo
 */

class HistoryAdapter internal constructor(
        private var data: List<BarangHarga>?
) : RecyclerView.Adapter<HistoryAdapter.HistoryHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): HistoryHolder {
        val inflater = LayoutInflater.from(parent.context)
        val binding = ItemKomoditiBinding.inflate(inflater, parent, false)
        return HistoryHolder(binding)
    }

    override fun onBindViewHolder(holder: HistoryHolder, position: Int) {
        data?.let {
            val barang = it[position]
            holder.bind(barang)
        }
    }

    override fun getItemCount(): Int {
        return data?.size ?: 0
    }

    internal fun swapData(newData: List<BarangHarga>?) {
        if (newData != null) {
            data = newData
            notifyDataSetChanged()
        }
    }

    inner class HistoryHolder(private val binding: ItemKomoditiBinding) : RecyclerView.ViewHolder(binding.root) {

        fun bind(barangHarga: BarangHarga) {
            binding.catatan = barangHarga
            binding.executePendingBindings()
        }

    }
}